poet/poet.py

```diff
@@ -185,14 +198,17 @@ def get_checksum(self) -> str:
             raise e
         
         try:
-            tar_ball = [
+            tar_ball = next(iter([
                 asset for asset in response["assets"] if ".tar.gz" in asset["name"]
-            ][0]
-            return tar_ball["hashes"]["SHA-256"]
+            ]))
+        except StopIteration as stop_iteration:
+            raise CodeArtifactMetadataException("Could not find tarball asset") from stop_iteration
```

Comment:
I wonder if you could be having issues with paging, and that's why you can't find certain assets
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/codeartifact.html#CodeArtifact.Paginator.ListPackageVersionAssets
I just noticed you aren't using the paginator for `list_package_version_assets`, and so in some cases you won't get all the results.

Do you have an example error that you're trying to address with this PR, or have you verified it isn't a pagination issue?

Reply:
`pip` itself does not upload enough assets to need pagination. When investigating some of the `404` errors when looking for the needed `tar.gz` files, some of the more distributed and older dependencies do not publish this file to Code Artifact. They do however sometimes have a `zip` file which I would like to stay away from because I want to stay uniform with the types of files being used in the formula.

Reply:
At some point, we may want pagination, but I am not seeing it be an immediate threat and have created a ticket for it: https://symbiota.atlassian.net/browse/DO-3864